src/Avalonia.Input/InputElement.cs

```diff
@@ -397,19 +397,23 @@ public GestureRecognizerCollection GestureRecognizers
         /// <summary>
         /// Focuses the control.
         /// </summary>
-        public void Focus()
-        {
-            FocusManager.Instance?.Focus(this);
-        }
+        public bool Focus() => this.GetFocusManager()?.Focus(this) == true;
+
+        /// <summary>
+        /// Focuses the control.
+        /// <param name="method">The method by which focus was changed.</param>
+        /// <param name="modifiers">Any input modifiers active at the time of focus.</param>
+        /// </summary>
+        public bool Focus(NavigationMethod method, InputModifiers modifiers = InputModifiers.None) => this.GetFocusManager()?.Focus(this, method, modifiers) == true;
 
         /// <inheritdoc/>
         protected override void OnDetachedFromVisualTreeCore(VisualTreeAttachmentEventArgs e)
         {
+            var focusManager = this.GetFocusManager();
```

Comment:
Nit: local variable not really needed here, and the call to `GetFocusManager` can be moved into the `if`.